src/peer/node.js

```diff
@@ -270,8 +278,57 @@ class MediachainNode {
     throw new Error('Local statement db not implemented!')
   }
 
-  data (keys: Array<string>): Array<DataResultMsg> {
-    throw new Error('Local datastore not implemented!')
+  putData (...vals: Array<Object | Buffer>): Promise<Array<string>> {
+    return Promise.all(
+      vals.map(val => this.datastore.put(val))
+    )
+  }
+
+  data (...keys: Array<string>): Promise<Array<DataObjectMsg>> {
+    const valuePromises = keys.map(k => this.datastore.get(k, {returnRawBuffer: true}))
+    return Promise.all(valuePromises)
+      .then(vals => {
+        const kvs = zip(keys, vals)
+        return kvs.map(([key, data]) => ({
+          key,
+          data
+        }))
+      })
+  }
+
+  dataHandler (protocol: string, conn: Connection) {
+    pull(
+      // read data request from stream and decode
+      conn,
+      protoStreamDecode(pb.node.DataRequest),
+
+      // make sure we close the stream when the node stops
+      this.p2p.newAbortable(),
+
+      // convert each request into a stream of keys, followed by a StreamEnd message object
+      pull.map(req => [...req.keys, {end: {}}]),
+      pull.flatten(),
+
+      // fetch values from the datastore, returning them in the same order as the input
+      paramap((keyOrEnd: string | Object, callback) => {
```

Comment:
a protocol test suite would be awesome.  maybe lets try to get that together after the break?